WalletWasabi/Wallets/Exchange/ExchangeRateUpdater.cs

```diff
@@ -2,38 +2,30 @@
 using System.Threading;
 using System.Threading.Tasks;
 using WalletWasabi.Bases;
-using WalletWasabi.Logging;
 using WalletWasabi.Services;
-using WalletWasabi.WebClients;
 
 namespace WalletWasabi.Wallets.Exchange;
 
 public class ExchangeRateUpdater : PeriodicRunner
 {
-	private readonly Func<string> _exchangeRateProviderGetter;
 	private readonly EventBus _eventBus;
 	private readonly ExchangeRateProvider _provider;
-	private readonly UserAgentPicker _userAgentPicker;
 	public decimal UsdExchangeRate { get; private set; }
 
-	public ExchangeRateUpdater(TimeSpan period, Func<string> exchangeRateProviderGetter,
-		IHttpClientFactory httpClientFactory, EventBus eventBus)
+	public ExchangeRateUpdater(TimeSpan period, ExchangeRateProvider exchangeRateProvider, EventBus eventBus)
 		: base(period)
 	{
-		_provider = new ExchangeRateProvider(httpClientFactory);
-		_exchangeRateProviderGetter = exchangeRateProviderGetter;
-		_userAgentPicker = UserAgent.GenerateUserAgentPicker(false);
+		_provider = exchangeRateProvider;
 		_eventBus = eventBus;
 	}
 
 	protected override async Task ActionAsync(CancellationToken cancellationToken)
 	{
-		var newExchangeRate = await _provider.GetExchangeRateAsync(_exchangeRateProviderGetter(), _userAgentPicker(), cancellationToken).ConfigureAwait(false);
+		var newExchangeRate = await _provider(cancellationToken).ConfigureAwait(false);
 		if (newExchangeRate.Rate != UsdExchangeRate && newExchangeRate.Rate > 0m)
 		{
 			UsdExchangeRate = newExchangeRate.Rate;
 			_eventBus.Publish(new ExchangeRateChanged(newExchangeRate.Rate));
-			Logger.LogInfo($"Fetched exchange rate from {_exchangeRateProviderGetter()}: {newExchangeRate.Rate}.");
```

Comment:
this was too hard to keep?

I think it's good to have at least one way to know where the exchange rate is being fetched from.

Reply:
You're right. I will do it.

Reply:
Implemented here: https://github.com/WalletWasabi/WalletWasabi/pull/14073